File: lab4/backend/api/schemas/center.py
from pydantic import BaseModel, ConfigDict
from typing import Optional

class CenterRead(BaseModel):
    id: int 
    blocks: Optional[int] = None
    rebounds: Optional[int] = None
    blocks_per_game: Optional[float] = None
    rebounds_per_game: Optional[float] = None

    model_config = ConfigDict(from_attributes=True)

class CenterCreate(BaseModel):
    blocks: Optional[int] = None
    rebounds: Optional[int] = None
    blocks_per_game: Optional[float] = None
    rebounds_per_game: Optional[float] = None

    model_config = ConfigDict(from_attributes=True)
    
class CenterUpdate(BaseModel):
    id: int
    blocks: Optional[int] = None
    rebounds: Optional[int] = None
    blocks_per_game: Optional[float] = None
    rebounds_per_game: Optional[float] = None

    model_config = ConfigDict(from_attributes=True)

File: lab4/gateway/internal/models/center.go
package models

type Center struct {
	ID              int     `json:"id"`
	Blocks          *int    `json:"blocks,omitempty"`
	Rebounds        *int    `json:"rebounds,omitempty"`
	BlocksPerGame   float64 `json:"blocks_per_game"`
	ReboundsPerGame float64 `json:"rebounds_per_game"`
}
